packages/foam-vscode/src/core/utils/path.ts

```diff
@@ -0,0 +1,102 @@
+import { CharCode } from '../common/charCode';
+import { posix } from 'path';
+import { promises, constants } from 'fs';
+
+export function asPath(value: string): [string, string] {
+  let authority: string;
+  if (isUNCShare(value)) {
+    [value, authority] = parseUNCShare(value);
+  }
+  if (value[0] === '/' && hasDrive(value, 1)) {
+    value = '/' + value[1].toUpperCase() + value.substr(2);
+  }
+  if (hasDrive(value, 0)) {
+    value = '/' + value[0].toUpperCase() + value.substr(1).replace(/\\/g, '/');
+  }
+  return [value, authority];
+}
+
+export function toFsPath(path: string, authority?: string): string {
```

Comment:
next to this one we should have `fromFsPath`

Reply:
See comment above.